File: explorer-backend/src/Models/API/SimplifiedBlock.cs
using explorer_backend.Models.Data;

namespace explorer_backend.Models.API;

public class SimplifiedBlock
{
    public int Height { get; set; }
    public int Size { get; set; }
    public int Weight { get; set; }
    public BlockType ProofType { get; set; }
    public long Time { get; set; }
    public long MedianTime { get; set; }

    public int TxCount { get; set; }
}

File: explorer-frontend/models/API/SimplifiedBlock.ts
export enum BlockType {
    UNKNOWN = 0,
    POW_X16RT = 1,
    POW_ProgPow = 2,
    POW_RandomX = 3,
    POW_Sha256D = 4,
    POS = 5
}

export interface SimplifiedBlock {
    height: number;
    size: number;
    weight: number;
    proofType: BlockType;
    time: number;
    medianTime: number;
    txCount: number;
}